### PR TITLE
feat: predict metric

### PR DESCRIPTION
**Pull request by cabreraalex**
Fix EOT-61

### CODE REVIEW

File: src/lib/components/popups/ProjectSettingsPopup.svelte [hunk 1]
```diff
@@ -97,6 +100,7 @@
 					onchange={async () => {
 						const projectRes = await toggleProjectLabels(
 							project.id,
+							undefined,
```
Comment (Sparkier) [inline src/lib/components/popups/ProjectSettingsPopup.svelte:103]: why don't we pass the prompt here as well?

File: src/lib/components/Instances/InstanceTable.svelte [hunk 2]
```diff
@@ -103,6 +103,7 @@
 						project.show_labels = true;
 						const projectRes = await toggleProjectLabels(
 							project.id,
+							prompt.prompt,
```
Comment (Sparkier) [inline src/lib/components/Instances/InstanceTable.svelte:106]: I'd send the whole prompt. We can infer from meta vars as well. E.g. if it's JSON we probably want to just compare json in a metric.

File: src/lib/server/metrics/predictMetric.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+You can pick from the following metrics:
+"chrf" - measures the bi-gram overlap of two text spans. Good for generative tasks such as summarization, text generation, etc.
+"exact match" - measures if prediction and label are identical. Good for logical or exact tasks such as data parsing
+"fuzzy match" - measures if the prediction has at least part of the label.
+
+Return JSON with the key "output" and the string of the metric.
+`;
+
+export async function predictMetric(prompt: string | undefined): Promise<string> {
```
Comment (Sparkier) [inline src/lib/server/metrics/predictMetric.ts:15]: As mentioned, would maybe check prompt output type before querying a model about it.

**Review by Sparkier — CHANGES_REQUESTED** [review]

File: src/lib/components/Instances/InstanceTable.svelte (hunk 2 above)
Comment (cabreraalex) [reply src/lib/components/Instances/InstanceTable.svelte:106]: good idea

**Review by Sparkier — APPROVED** [review]

**Pull request merged**